public class java.util.logging.ErrorManager extends java.lang.Object
{
    public static final int CLOSE_FAILURE;
    private static final java.lang.String[] FAILURES;
    public static final int FLUSH_FAILURE;
    public static final int FORMAT_FAILURE;
    public static final int GENERIC_FAILURE;
    public static final int OPEN_FAILURE;
    public static final int WRITE_FAILURE;
    private boolean called;

    static void <clinit>()
    {
        java.lang.String[] $r0;

        <java.util.logging.ErrorManager: int WRITE_FAILURE> = 1;

        <java.util.logging.ErrorManager: int OPEN_FAILURE> = 4;

        <java.util.logging.ErrorManager: int GENERIC_FAILURE> = 0;

        <java.util.logging.ErrorManager: int FORMAT_FAILURE> = 5;

        <java.util.logging.ErrorManager: int FLUSH_FAILURE> = 2;

        <java.util.logging.ErrorManager: int CLOSE_FAILURE> = 3;

        $r0 = newarray (java.lang.String)[6];

        $r0[0] = "GENERIC_FAILURE";

        $r0[1] = "WRITE_FAILURE";

        $r0[2] = "FLUSH_FAILURE";

        $r0[3] = "CLOSE_FAILURE";

        $r0[4] = "OPEN_FAILURE";

        $r0[5] = "FORMAT_FAILURE";

        <java.util.logging.ErrorManager: java.lang.String[] FAILURES> = $r0;

        return;
    }

    public void <init>()
    {
        java.util.logging.ErrorManager r0;

        r0 := @this: java.util.logging.ErrorManager;

        specialinvoke r0.<java.lang.Object: void <init>()>();

        return;
    }

    public void error(java.lang.String, java.lang.Exception, int)
    {
        java.util.logging.ErrorManager r0;
        java.lang.String r1, $r7, $r11, $r12, $r19, $r22;
        java.lang.Exception r2;
        int i0;
        java.lang.Throwable r3, $r24;
        boolean $z0;
        java.lang.StringBuilder $r4, $r8, $r9, $r13, $r15, $r16, $r17, $r18, $r20, $r23;
        java.io.PrintStream $r5, $r14, $r21;
        java.lang.Class $r6;
        java.lang.String[] $r10;

        r0 := @this: java.util.logging.ErrorManager;

        r1 := @parameter0: java.lang.String;

        r2 := @parameter1: java.lang.Exception;

        i0 := @parameter2: int;

        entermonitor r0;

     label1:
        $z0 = r0.<java.util.logging.ErrorManager: boolean called>;

        if $z0 == 0 goto label3;

        exitmonitor r0;

     label2:
        return;

     label3:
        r0.<java.util.logging.ErrorManager: boolean called> = 1;

        exitmonitor r0;

     label4:
        $r5 = <java.lang.System: java.io.PrintStream err>;

        $r4 = new java.lang.StringBuilder;

        specialinvoke $r4.<java.lang.StringBuilder: void <init>()>();

        $r6 = virtualinvoke r0.<java.lang.Object: java.lang.Class getClass()>();

        $r7 = virtualinvoke $r6.<java.lang.Class: java.lang.String getName()>();

        $r8 = virtualinvoke $r4.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>($r7);

        $r9 = virtualinvoke $r8.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>(": ");

        $r10 = <java.util.logging.ErrorManager: java.lang.String[] FAILURES>;

        $r11 = $r10[i0];

        $r13 = virtualinvoke $r9.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>($r11);

        $r12 = virtualinvoke $r13.<java.lang.StringBuilder: java.lang.String toString()>();

        virtualinvoke $r5.<java.io.PrintStream: void println(java.lang.String)>($r12);

        if r1 == null goto label5;

        $r21 = <java.lang.System: java.io.PrintStream err>;

        $r18 = new java.lang.StringBuilder;

        specialinvoke $r18.<java.lang.StringBuilder: void <init>()>();

        $r20 = virtualinvoke $r18.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>("Error message - ");

        $r23 = virtualinvoke $r20.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>(r1);

        $r22 = virtualinvoke $r23.<java.lang.StringBuilder: java.lang.String toString()>();

        virtualinvoke $r21.<java.io.PrintStream: void println(java.lang.String)>($r22);

     label5:
        if r2 == null goto label9;

        $r14 = <java.lang.System: java.io.PrintStream err>;

        $r15 = new java.lang.StringBuilder;

        specialinvoke $r15.<java.lang.StringBuilder: void <init>()>();

        $r17 = virtualinvoke $r15.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>("Exception - ");

        $r16 = virtualinvoke $r17.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.Object)>(r2);

        $r19 = virtualinvoke $r16.<java.lang.StringBuilder: java.lang.String toString()>();

        virtualinvoke $r14.<java.io.PrintStream: void println(java.lang.String)>($r19);

        return;

     label6:
        $r24 := @caughtexception;

        r3 = $r24;

     label7:
        exitmonitor r0;

     label8:
        throw r3;

     label9:
        return;

        catch java.lang.Throwable from label1 to label2 with label6;
        catch java.lang.Throwable from label3 to label4 with label6;
        catch java.lang.Throwable from label7 to label8 with label6;
    }
}
